### ISSUE TITLE
Upgrade to magnus 0.7.x → Threads, JS daemon, and better scheduler control

### ISSUE THREAD

**Issue by eliias** [eliias/isorun#299]
The Magnus 0.7.0 release added support for Ruby Threads, Mutex, and Fibers. This is a good time to revisit our usage of the fork that supports releasing the GVL, and to upgrade it to work with 0.7.x. We currently use the fork to allow us to unblock the Rails process during the time we spend computing in v8/Rust. There is other a lot of bad code that allow us to interop between Ruby and v8, and this should be improved:

1. It is currently unclear what the actual cost is to create a v8 “worker” every time we call a JS function.
2. It would be nice to have a pre-warmed set of v8 threads/Ruby process (configurable), instead of one worker/process. This allows users to balance memory/performance trade-offs in a better way, and maybe enables some form of concurrency for function calls (maybe helpful for concurrent rendering of React apps in templates).
